raiden/transfer/channel.py

```diff
@@ -892,6 +903,10 @@ def is_valid_withdraw_confirmation(
 
     result: SuccessOrError
 
+    withdraw_exists = withdraw.total_withdraw in [
+        withdraw_state.total_withdraw for withdraw_state in channel_state.our_state.withdraws
```

Comment:
Again, I think we should use a data structure for this, it is silly to do a linear search for something like this.

Reply:
I can switch this to a `dict` with `total_withdraw` as the key. Since `total_withdraw` is monotonic, it'll be guaranteed that it's unique in this context to serve as the key. Agreed?